git-remote-ipfs/test/e2e/Main.hs

```diff
@@ -0,0 +1,79 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Main where
+
+import           Control.Exception.Safe (throwString)
+import           Control.Monad.Reader
+import qualified Data.ByteString.Char8 as Bytes
+import           Data.Foldable (foldlM)
+import           Data.Maybe (maybeToList)
+import           Data.Tagged (untag)
+import qualified Data.Text as Text
+import           Data.Time (getZonedTime)
+import           GHC.Stack (HasCallStack)
+import           Git
+import           Git.CmdLine
+import           Git.Libgit2 (LgRepo, lgFactory)
+import           System.FilePath ((</>))
+import           System.IO.Temp (withSystemTempDirectory)
+
+import           Test.Tasty
+import           Test.Tasty.HUnit
+
+main :: IO ()
+main = defaultMain $
+    testCase "Simple push, clone works" $
+        withSystemTempDirectory "git-remote-ipfs-e2e" testPushCloneSimple
+
+testPushCloneSimple :: HasCallStack => FilePath -> IO ()
+testPushCloneSimple root = do
+    let repoPath  = root </> "simple" </> ".git"
+    let clonePath = root </> "simple-clone"
+
+    withRepository' lgFactory (ropts repoPath True) $
+        maybe (throwString "headless history")
+              (updateReference "refs/heads/master" . RefObj . untag . commitOid)
+            =<< simpleHistory
+
+    url <-
+        withRepository' cliFactory (ropts repoPath False) $ do
+            runGit_ ["remote", "add", "ipfs", "ipfs://"]
+            runGit_ ["push", "--quiet", "ipfs", "master"]
+            Text.stripEnd <$> runGit  ["remote", "get-url", "ipfs"]
+
+    withRepository' cliFactory (ropts (clonePath </> ".git") False) $ do
```

Comment:
Perhaps is outside the scope of these end-2-end tests, but is there any `Assertion` we could meaningfully perform at this stage, after we exist the second `withRepository'` block?

Reply:
See below - the assertion is `git diff -q` exiting `0`